### ISSUE TITLE
isssu

### ISSUE THREAD

**Issue by aghili371** [OrchardCMS/OrchardCore#3635]

**Comment by hishamco** [comment]
Please screen shoot

**Comment by hishamco** [comment]
Please screen shoot

**Comment by sebastienros** [comment]
I think he's talking about the picker itself. When you edit a content item.

**Comment by hishamco** [comment]
Yep, but I'm not sure if it's built with bootstrap, because this is something different if I'm not wrong

**Comment by hishamco** [comment]
@sebastienros is the component using bootstrap or jquery-ui or native one?

**Comment by aghili371** [comment]
![Capture](https://user-images.githubusercontent.com/24512878/57804923-1e444d00-775c-11e9-8d5d-c6990715bc86.JPG)
i can't make screen shoot. when you want to select date , it doesn't show Persian calendar.

**Comment by sebastienros** [comment]
maybe there is a default in-browser editor we could use as a custom editor.

**Comment by hishamco** [comment]
That's a default browser editor we use custom one that support localization for instance jquery ui datepicker supports many languages including Arabic, Farsi .. etc

**Comment by sebastienros** [comment]
Looks good to use the jquery one then, ideally as a custom editor at first.

**Comment by hishamco** [comment]
This will need the following:

- Add jQuery UI DataPicker localization script to be added into`OrchardCore.Resources`
- Add a custom editor for the `DateField` which may I need your help on this, because I didn't work with the editors yet
- Reference the added resource into the admin theme

FYI there 're two options for datetime picker localization script:
1- Script per language which is better in term of Perf
2- Scripts for all the supported languages (which is the preferred option for us)

**Comment by hishamco** [comment]
After little investigation seems jquery-ui support datetime localization only. BTW I found another cool library that supports localization & calendars too but it needs more investigation, so for that I'm suggesting the following:

1- Add jQuery-UI date picker as a custom editor as @sebastienros mentioned before
2- Integrate the new date picker, with specifying the calendar and language as settings (which will fix #3664)
3- Mapping the date picker calendars with the one that we already have and support

FYI: the priority for step 1, other step may added in 1.1.0 as improvements, but if we have time we can do it in 1.0.0

**Comment by hishamco** [comment]
@aghili371 the PR #3707 added a localized version of datetime picker, @sebastienros can we make this open until we finish other steps, or shall we make the improvements after `1.0.0` milestone?

**Comment by aghili371** [comment]
it still doesn't work.

**Comment by hishamco** [comment]
@aghili371 the localized date-picker should should the name of the Persian days, months and years if the culture set to `fa-IR`, don't confuse with Persian calendar which is still not integrated with the date-picker

**Comment by Skrypt** [comment]
Create a SiteSettingsCalendarSelector that would override the DefaultCalendarSelector that uses the calendar that comes from the current culture.

https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore/OrchardCore/Localization/DefaultCalendarSelector.cs#L14

**Comment by hishamco** [comment]
As we discussed in gitter, if date picker should uses the `SiteSettingsCalendarSelector` unless overriding need. For instance if the `SiteSettingsCalendarSelector` provides `GregorianCalendar`, may I will override that and use `HijriCalendar` for specific date pickers

**Comment by sebastienros** [comment]
- should it be an option of the field to chose which picker to use? 
- should the date picker always match the current calendar, at least with its default settings, (if not point 1)?

I see online that there are different date pickers based on the calendar. Maybe we should have an international date picker that can adapt to the current culture, or/and provide custom editor for the date field that lets us select which picker to use. In the end the date would still be stored with a DateTime value.

**Comment by hishamco** [comment]
@sebastienros can reschedule the enhancements that I mentioned in my comment https://github.com/OrchardCMS/OrchardCore/issues/3635#issuecomment-496006787 to target `1.1`
